lrpar/cttests/src/lib.rs

```diff
@@ -88,3 +91,51 @@ fn test_calc_multitypes() {
     let (res, _errs) = calc_multitypes_y::parse(&mut lexer);
     assert_eq!(res, Some(Ok(3)));
 }
+
+#[test]
+fn test_span() {
+    let lexerdef = span_l::lexerdef();
+    let mut lexer = lexerdef.lexer("2+3");
+    match span_y::parse(&mut lexer) {
+        (Some(ref spans), _) if spans == &vec![(0, 1), (0, 1), (2, 3), (2, 3), (2, 3), (0, 3)] => {
+            ()
+        }
+        _ => unreachable!()
+    }
+
+    let mut lexer = lexerdef.lexer("2+3*4");
+    match span_y::parse(&mut lexer) {
+        (Some(ref spans), _)
+            if spans
+                == &vec![
+                    (0, 1),
```

Comment:
It's scary how it didn't even confuse me why some of these appear multiple times, because my head had already parsed the above input and made the `P->T` reductions.

Reply:
Ha! It took me a few times to get my head around it!